### PR TITLE
fix(nuxt): account for delay ≤ 0 in `useCookie`

### PR DESCRIPTION
**Pull request by DamianGlowala**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR ensures `maxAge` is respected for negative values and 0, which is a falsy value. For delay less than or equal to 0, `setTimeout` would default it to 1, whereas this PR sets it to `undefined` immediately instead.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#max-agenumber

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/app/composables/cookie.ts [hunk 1]
```diff
@@ -32,10 +32,11 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
...
-  if (opts.maxAge) {
+
+  if (opts.maxAge !== undefined) {
     delay = opts.maxAge * 1000 // convert to ms for setTimeout
   } else if (opts.expires) {
-    // getTime() already return time in ms
+    // getTime() already returns time in ms
     delay = opts.expires.getTime() - Date.now()
   }
   // use customRef if on client side otherwise use basic ref
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/cookie.ts:42]: what about just checking that `delay >= 1` on L43 so we don't use a `cookieRef` in that case?

File: packages/nuxt/src/app/composables/cookie.ts [hunk 2]
```diff
@@ -32,14 +32,15 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
   const cookies = readRawCookies(opts) || {}
 
   let delay: number | undefined
-  if (opts.maxAge) {
+
+  if (opts.maxAge !== undefined) {
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/cookie.ts:36]: I think you're accounting for 0 here but we don't do anything with it if so, and it doesn't handle null this way.

```suggestion
  if (opts.maxAge) {
```

Comment (DamianGlowala) [reply packages/nuxt/src/app/composables/cookie.ts:36]: image

Type def disallows `null` though, do we need to handle it?

Also, the initial implementation accounted for 0 and set the value to `undefined`. Reverted to it with some minor tweaks. I kind of prefer letting the `cookieRef` handle the delay _regardless of its value_. Open to your thoughts!

Comment (DamianGlowala) [reply packages/nuxt/src/app/composables/cookie.ts:36]: The below sentence is a sole reason why I thought about making this PR:
image

Comment (danielroe) [reply packages/nuxt/src/app/composables/cookie.ts:36]: No, that should definitely be the case, but I'm not sure that `cookieRef` will do that as-is (or is needed to do that).

Comment (danielroe) [reply packages/nuxt/src/app/composables/cookie.ts:36]: Shall we instead add a handler for that so we do it directly after we initialise the ref?

Comment (DamianGlowala) [reply packages/nuxt/src/app/composables/cookie.ts:36]: Just pushed an update which I hope is closer to the desirable solution. Let me know your thoughts

**Review by danielroe — APPROVED** [review]

**Pull request merged**